_adr/68/index.adoc

```diff
@@ -0,0 +1,91 @@
+title: "Colocating managed services on the same data plane OSD fleet"
+status: "Draft"
+authors:
+  - "Keith Wall"
+  - "Tom Bentley"
+  - "David Ffrench"
+---
+
+## Context and Problem Statement
+
+As Red Hat starts to offer additional managed services, a portion of services will follow the bin-packed data plane deployment model which is used in RHOSAK. Each new data plane OSD cluster added to the fleet requires 3 master nodes and 3 infra nodes for a multi-az cluster. If each new bin-packed managed service deploys their own data plane OSD fleet, Red Hat's underlying infrastructure cost will be higher.
+
+If Red Hat does not address this problem, the implication is the cost to customers of new managed services will be higher due to the additional master and infra nodes causing an increase in data plane infrastructure cost.
+
+## Goals
+* Identify how different managed services can be co-located on the same data plane fleet
+* Outline network and capacity isolation between managed services
+
+## Non-goals
+* How installation of managed services will be initiated on the data plane fleet
+* Outline how different SRE teams manage separate managed services on the same data plane fleet
+* Determine the ideal node type (e.g. m5.xlarge) for new managed services
+* Outline multi-az deployment model of new managed services
+
+## Stakeholders
+* Eng
+* BU
+* SRE
+
+## Current Architecture
+
+* We don’t currently have a second managed service with a deployment model which requires a data plane OSD cluster fleet.
+
+* RHOSAK's data plane OSD clusters are created manually and registered with kas-fleet-manager which then installs all the components required such as the fleet-shard-operator, Strimzi operator and observability operator. Only once these components have installed successfully is the data plane OSD cluster ready to start assigning Kafka instances.
+* All Kafka instances are scheduled to the compute nodes on the OSD clusters, there are no taints or tolerations currently in place. 
+
+## Proposed Architecture
+
+### Capacity Isolation
+
+A new machine pool will be created for each new managed service. A machine pool is a group of worker nodes in a cluster that have the same configuration, providing ease of management.
+
+This allows each new managed service to determine their own instance type that suits their needs. For example, M5 (general purpose), C5 (CPU optimised), R5 (Memory optimised) and at the size they would like.
+
+All machine pools will be created through the OpenShift Cluster Manager (OCM) API, not directly on the OpenShift Dedicated (OSD) clusters. Each machine pool will be created with a `NoSchedule` taint which will require all pods scheduled for that managed service to include the toleration.
+
+While outside the scope of this ADR, it is worth noting that with a shared data plane fleet, a new micro service in the control plane would be ideal to control scaling machine pools, creation and deletion of data plane OSD clusters.
```

Comment:
I suppose this is talking about the [Terraforming API proposal](https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/issues/4), correct? @davidffrench Maybe we can add a link.